recipes/mp-units/2.2.0/conanfile.py

```diff
@@ -0,0 +1,266 @@
+
+
+class MPUnitsConan(ConanFile):
+    name = "mp-units"
+    homepage = "https://github.com/mpusz/mp-units"
+    description = "The quantities and units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "cxx_modules": ["auto", True, False],
+        "std_format": ["auto", True, False],
+        "string_view_ret": ["auto", True, False],
+        "no_crtp": ["auto", True, False],
+        "contracts": ["none", "gsl-lite", "ms-gsl"],
+        "freestanding": [True, False],
+    }
+    default_options = {
+        "cxx_modules": "auto",
+        "std_format": "auto",
+        "string_view_ret": "auto",
+        "no_crtp": "auto",
+        "contracts": "gsl-lite",
+        "freestanding": "False",
+    }
+    tool_requires = "cmake/[>=3.29]"
+    package_type = "header-library"
+    no_copy_source = True
+
+    @property
+    def _feature_compatibility(self):
+        return {
+            "minimum_support": {
+                "std": "20",
+                "compiler": {
+                    "gcc": "12",
+                    "clang": "16",
+                    "apple-clang": "15",
+                    "msvc": "",
+                },
+            },
+            "std_format": {
+                "std": "20",
+                "compiler": {
+                    "gcc": "13",
+                    "clang": "17",
+                    "apple-clang": "",
+                    "msvc": "",
+                },
+            },
+            "cxx_modules": {
+                "std": "20",
+                "compiler": {"gcc": "14", "clang": "17", "apple-clang": "", "msvc": ""},
+            },
+            "static_constexpr_vars_in_constexpr_func": {
+                "std": "23",
+                "compiler": {"gcc": "13", "clang": "17", "apple-clang": "", "msvc": ""},
+            },
+            "explicit_this": {
+                "std": "23",
+                "compiler": {
+                    "gcc": "14",
+                    "clang": "18",
+                    "apple-clang": "",
+                    "msvc": "",
+                },
+            },
+        }
+
+    @property
+    def _option_feature_map(self):
+        return {
+            "std_format": "std_format",
+            "cxx_modules": "cxx_modules",
+            "string_view_ret": "static_constexpr_vars_in_constexpr_func",
+            "no_crtp": "explicit_this",
+        }
+
+    def _check_feature_supported(self, name, feature_name=name):
+        compiler = self.settings.compiler
+        cppstd = compiler.get_safe("cppstd", default_cppstd(self))
```

Comment:
This is a bit weird, why use the default_cppstd here as the default? I'm guessing it's for Conan 1 compatibility right?

Reply:
Is cppstd always provided (even if missing in the profile)? Do we still care about Conan 1? If not, we can probably remove the `test_v1_package` directory.

Reply:
Let me check with the team, we probably can mark this recipe as only v2-compatible, but as for your question:

In Conan 1, no, the cppstd value is not required to be set, old Conan does not model the cppstd too well

As for Conan 2, yes! The cppstd value is meant to always be defined :)